receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -67,6 +69,13 @@ func newMetricFamilyPdata(metricName string, mc MetadataCache, intervalStartTime
 			metadata.Metric = familyName
 			metadata.Type = textparse.MetricTypeUnknown
 		}
+	} else if !ok && isInternalMetric(metricName) {
+		metadata = defineInternalMetric(metricName, metadata, logger)
+	}
+
+	mtype := convToPdataMetricType(metadata.Type)
+	if mtype == pdata.MetricDataTypeNone {
+		logger.Debug(fmt.Sprintf("Invalid metric : %s %+v", metricName, metadata))
```

Comment:
MetricName and metadata should fields not a dynamic string to allow for easier searching in logging systems.

Reply:
I also question the value of having the log statement here?

Reply:
It's in the original code https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0040189a514c9a28a8fb33b49ce257f4fd327320/receiver/prometheusreceiver/internal/metricfamily.go#L71-L74

this code is a port. @MovieStoreGuy I appreciate the enthusiasm but the context here is that I am porting over code, which requires careful parity changes; you can file bugs for improvements but these PRs am sending are for porting over code.